instrumentation/struts-2/src/main/java/io/opentelemetry/javaagent/instrumentation/struts2/ActionInvocationAdvice.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.struts2;
+
+import static io.opentelemetry.javaagent.instrumentation.struts2.Struts2Tracer.TRACER;
+
+import com.opensymphony.xwork2.ActionInvocation;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.javaagent.instrumentation.api.Java8BytecodeBridge;
+import net.bytebuddy.asm.Advice;
+
+public class ActionInvocationAdvice {
```

Comment:
It looks like struts has the concept of interceptors

https://struts.apache.org/core-developers/interceptors.html

Is it possible to define an interceptor and register it, instead of instrumenting the action invocation directly? While it wouldn't need to be in this PR, it would make it much easier to separate out library, nonagent instrumentation.

Reply:
Yes, interceptors definitely can be used for a library. Moved current instrumentation to `javaagent` sub-project to accommodate for the library version. Which I will do as a separate PR.